File: core/SyncPlatform/src/Services/Roadmap/Roadmap.Domain/Models/RecoveryProfile.cs
using Roadmap.Domain.Models;

namespace Roadmap.Domain.Models;

public class RecoveryProfile : BaseMongoEntity
{
    public Guid UserId { get; set; }

    public int CurrentRecoveryScore { get; set; }

    public int FatigueLevel { get; set; }

    public int SleepRecoveryScore { get; set; }

    public int MuscleSorenessScore { get; set; }

    public int CnsFatigueScore { get; set; }

    public string RecommendedTrainingIntensity { get; set; } = string.Empty;

    public int RecommendedWorkoutDuration { get; set; }
}

File: ai-ecosystem/agentic-lifestyle/src/sync_agent/domain/schemas/roadmap/recovery_profile.py
"""
Roadmap.Domain.Models.RecoveryProfile — JIT context for Workout Action node.
Not in Task 1 list but required by .cursorrules (CnsFatigueScore, MuscleSorenessScore).
"""

from uuid import UUID

from sync_agent.domain.schemas.common import MongoEntityFields


class RecoveryProfile(MongoEntityFields):
    """Maps C# RecoveryProfile exactly."""

    user_id: UUID
    current_recovery_score: int
    fatigue_level: int
    sleep_recovery_score: int
    muscle_soreness_score: int
    cns_fatigue_score: int
    recommended_training_intensity: str
    recommended_workout_duration: int
